src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -453,9 +450,6 @@ private void LookupExtensionMethodsInSingleBinder(ExtensionMethodScope scope, Lo
                 resultWithSuffix = LookupResult.GetInstance();
                 this.LookupSymbolsOrMembersInternal(resultWithSuffix, qualifierOpt, name + "Attribute", arity, basesBeingResolved, options, diagnose, ref useSiteDiagnostics);
                 resultWithSuffixIsViable = IsSingleViableAttributeType(resultWithSuffix, out symbolWithSuffix);
-
-                // Generic types are not allowed.
-                Debug.Assert(arity == 0 || !result.IsMultiViable);
             }
 
             if (resultWithoutSuffixIsViable && resultWithSuffixIsViable)
```

Comment:
Feels like we need to do something in this case too, to maintain old behavior. What if both results (with and without suffix) are viable, but one is generic, the other is not generic and the LangVersion is 8. Then it should behave as-if we only had one viable result. 
Let's make sure to test those permutations, if not already covered.